packages/rest/test/test-resource.ts

```diff
@@ -83,6 +83,46 @@ describe("rest: resources", () => {
     ]);
   });
 
+  it("resources: resources with parents must not have duplicate their parents' key names", async () => {
+    const [_, diagnostics] = await compileOperations(`
+      using Cadl.Rest.Resource;
+
+      namespace Things {
+        model Thing {
+          @key("thingId")
+          @segment("things")
+          id: string;
+        }
+
+        @parentResource(Thing)
+        model Subthing {
+          @key
+          @segment("subthings")
+          thingId: string;
+        }
+
+        @parentResource(Subthing)
+        model SubSubthing {
+          @key("thingId")
+          @segment("subsubthings")
+          subSubthingId: string;
+        }
+      }
+      `);
+
+    strictEqual(diagnostics.length, 2);
```

Comment:
Made a util here that verify the diagnostic for you
```
expectDiagnostics(diagnostics, [
  ... list of match

  {code: ..., message: ...}
])
```

Reply:
Fixed, thanks!